lib/Models/SdmxJsonCatalogItem.js

```diff
@@ -590,53 +601,84 @@ function buildRegionAndTimeColumns(item, dimensions) {
     }
 }
 
-function getIndexOfDimensionValueId(dimension, valueId) {
-    var result;
-    for (var i = 0; i < dimension.values.length; i++) {
-        if (dimension.values[i].id === valueId) {
-            result = i;
+// Sums an array, treating undefined's as 0 in the sum, but leaving undefined + undefined = undefined.
+function sumArray(array) {
+    return array.reduce(function(x, y) {
+        if (!defined(x)) {
+            return y;
         }
-    }
-    return result;
+        if (!defined(y)) {
+            return x;
+        }
+        return x + y;
+    }, 0);
```

Comment:
This might be neater as `array.filter(defined).reduce((x, y) => x + y, 0)` but it's a nitpick.